docker-entrypoint.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+	local var="$1"
+	local fileVar="${var}_FILE"
+	local def="${2:-}"
+	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+		echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+		exit 1
+	fi
+	local val="$def"
+	if [ "${!var:-}" ]; then
+		val="${!var}"
+	elif [ "${!fileVar:-}" ]; then
+		val="$(< "${!fileVar}")"
+	fi
+	export "$var"="$val"
+	unset "$fileVar"
+}
+
+# load secrets either from environment variables or files
+file_env 'DATABASE_URL'
+file_env 'GOTRUE_JWT_SECRET'
+file_env 'GOTRUE_SMTP_PASS'
+
```

Comment:
shouldn't this list of secrets be the entire list of env vars for gotrue?

Reply:
The list of variables are defined through docker secrets, I don't think it is necessary to include all the other non-secret variables. If we did that, it will be a bit strange to see those non-secret variables are placed in `/run/secrets/` folder.